services/worker-manager/src/providers/aws.js

```diff
@@ -271,8 +269,12 @@ class AwsProvider extends Provider {
       w.state = this.Worker.states.RUNNING;
     });
 
-    // return object that has expires field
-    return {expires: taskcluster.fromNow('96 hours')};
+    let expires = taskcluster.fromNow('96 hours');
+    if (worker.providerData.reregistrationDeadline) {
+      expires = new Date(worker.providerData.reregistrationDeadline);
```

Comment:
I suspect the providerData should contain the time delta, rather than an absolute time, as reregister will need to calculate a new deadline.

Reply:
Will do this when we add back reregister!